examples/rag/rag_agents.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+"""
+This example shows how to build an agent with RAG (backup by LlamaIndex)
+"""
+
+from typing import Optional
+from llama_index.core import SimpleDirectoryReader
```

Comment:
Use "try" and “expect ImportError" for llama_index and langchain libraries.

Reply:
updated